maxfw/tests/test_image_utils.py

```diff
@@ -0,0 +1,171 @@
+# Standard libs
+import io
+
+# Dependencies
+import pytest
+import numpy as np
+from PIL import Image
+
+# The module to test
+from maxfw.utils.image_utils import ImageProcessor, ToPILImage, Resize, Grayscale, Normalize, Standardize, Rotate
+
+# Initialize a test input file
+stream = io.BytesIO()
+Image.open('maxfw/tests/test_image.jpg').convert('RGBA').save(stream, 'PNG')
+test_input = stream.getvalue()
+
+
+def test_imageprocessor_read():
+    """Test the Imageprocessor."""
+
+    # Test with 4 channels
+    transform_sequence = [ToPILImage('RGBA')]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (678, 1024, 4)
+
+    # Test with 3 channels
+    transform_sequence = [ToPILImage('RGB')]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (678, 1024, 3)
+
+
+def test_imageprocessor_resize():
+    """Test the Imageprocessor's resize function."""
+
+    # Resize to 200x200
+    transform_sequence = [ToPILImage('RGBA'), Resize(size=(200, 200))]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (200, 200, 4)
+
+    # Resize to 2000x2000
+    transform_sequence = [ToPILImage('RGBA'), Resize(size=(2000, 2000))]
+    p = ImageProcessor(transform_sequence)
+    img_out = p.apply_transforms(test_input)
+    assert np.array(img_out).shape == (2000, 2000, 4)
+
+
+def test_imageprocessor_grayscale():
```

Comment:
The grayscale tests should have some test cases for the actual values. 

At least in the case of multiple channels, that they are the same (if that is the expected case)?

Reply:
Added in the next update!